W-12210401-xml-module-1.4.0-kt (#2185)
mulesoft/docs-connectors: xml/1.4/modules/ROOT/pages/xml-schema-validation.adoc

+
```
ERROR 2018-02-16 14:35:45,722 [[MuleRuntime].cpuIntensive.01: [SchemaValidationTestCase#extractErrorsUsingExpressions].extractErrorsFromException.CPU_INTENSIVE @411e886b] org.mule.runtime.core.internal.processor.LoggerMessageProcessor: At line: -1, column: -1 -> cvc-complex-type.2.4.a: Invalid content was found starting with element 'fail'. One of '{used}' is expected.
```
image::xml-validate-schema-3.png[The XML module manage validation error flow]

In the *Configuration XML* editor, the `<on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">` configuration looks like this:

[source,xml,linenums]
----
 <flow name="extractErrorsFromException">
    <try>
        <xml-module:validate-schema schemas="schema.xsd" />
        <error-handler>
            <on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">
                <foreach collection="#[error.errorMessage.payload]">
                    <logger level="ERROR" message="#['At line: $(payload.lineNumber), column: $(payload.columnNumber) -> $(payload.description)']" />
                </foreach>
            </on-error-propagate>
        </error-handler>
    </try>
</flow>
----

== Configure Schema Text Content

> In the following example, you configure the *Schema contents* field of the the *Validate schema* operation. The *Schema contents* field defines a direct text input of schema to perform a schema validation. 

Reviewer: ```suggestion
In the following example, you configure the *Schema contents* field of the *Validate schema* operation. Enter the schema text into this field to perform a schema validation. 
```

. In Studio, select the *Validate schema* operation from your flow.
. In the *Validate schema* operation configuration screen, set *Schema contents* to *Edit inline*.

image::xml-schema-content-1.png[Schema content field set to Edit inline in the Validate schema configuration screen]
[start=3]
. Click the plus sign to add a schema name and text.

> . In the *Schema content* window, set *Schema name* to define a schema name content, for example `schema1.xsd`.

Reviewer: ```suggestion
. In the *Schema content* window's *Schema name* field, enter a schema name, for example, `schema1.xsd`.
```

Reviewer: Trying to make the text less awkward.

> . Set *Schema text* to define the schema direct text content, for example:

Reviewer: In the *Schema text* field, enter the schema text content, for example:

[source,xml,linenums]
----
<?xml version = "1.0"?>
<xs:schema xmlns:xs = "http://www.w3.org/2001/XMLSchema">
   <xs:element name = 'class'>
      <xs:complexType>
         <xs:sequence>
             <xs:element name = 'student' type = 'StudentType' minOccurs = '0' 
                maxOccurs = 'unbounded' />
         </xs:sequence>
      </xs:complexType>
   </xs:element>

   <xs:complexType name = "StudentType">
      <xs:sequence>
         <xs:element name = "firstname" type = "xs:string"/>
         <xs:element name = "lastname" type = "xs:string"/>
         <xs:element name = "nickname" type = "xs:string"/>
         <xs:element name = "marks" type = "xs:positiveInteger"/>
      </xs:sequence>
      <xs:attribute name = 'rollno' type = 'xs:positiveInteger'/>
   </xs:complexType>             
</xs:schema>
----

> image::xml-schema-content-2.png[Schema content window with Schema name and Schema text fields]

Reviewer: ```suggestion
image::xml-schema-content-2.png[Schema content window with the Schema name and Schema text fields]
```

```
image::xml-validate-schema-3.png[The XML module manage validation error flow]

In the *Configuration XML* editor, the `<on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">` configuration looks like this:

[source,xml,linenums]
----
 <flow name="extractErrorsFromException">
    <try>
        <xml-module:validate-schema schemas="schema.xsd" />
        <error-handler>
            <on-error-propagate type="XML-MODULE:SCHEMA_NOT_HONOURED">
                <foreach collection="#[error.errorMessage.payload]">
                    <logger level="ERROR" message="#['At line: $(payload.lineNumber), column: $(payload.columnNumber) -> $(payload.description)']" />
                </foreach>
            </on-error-propagate>
        </error-handler>
    </try>
</flow>
----

== Configure Schema Text Content

In the following example, you configure the *Schema contents* field of the the *Validate schema* operation. Enter the schema text into this field to perform a schema validation.  

[NOTE]

> To add your schema content, you only provide either schema or schema content, but not both. You can either upload your file in the *Schemas* field, or manually add the direct schema text content in the *Schema contents* field.

Reviewer 2: I think you should also mention this in the parameters reference. (and please, check the required mark in the parameters)
